fetch.bs

```diff
@@ -3684,11 +3785,35 @@ steps:
 
  <li><p>Wait for either <var>response</var>'s <a for=response>body</a> to be null, or
  <var>response</var>'s <a for=response>body</a>'s <a for=body>stream</a> to be
- <a for=ReadableStream>closed</a> or <a for=ReadableStream>errored</a>, and then set
- <var>request</var>'s <a>done flag</a>.
- <!-- This is really bad and needs to be handled differently at some point. -->
+ <a for=ReadableStream>closed</a> or <a for=ReadableStream>errored</a>, and perform the following steps:
+ <ol>
+  <li><p>Set <var>request</var>'s <a>done flag</a>.
+   <!-- This is really bad and needs to be handled differently at some point. -->
+  <li>Call <a for=/>report resource timing</a> with <var>fetchParams</var> and <var>response</var>.
+ </ol>
 </ol>
 
+<p>To <dfn>report resource timing</dfn>, given <a for=/>fetch params</a> <var>fetchParams</var> and
+<a for=/>response</a> <var>response</var>, perform the following steps:
+<ol>
+  <li>Let <var>globalObject</var> be <var>fetchParams</var>'s <a for="fetch params">request</a>'s
+  <a for=request>client</a>'s <a for="environment settings object">global object</a>.
+  <li>Let <var>timingInfo</var> be <var>fetchParams</var>'s <a for="fetch params">timing info</a>.
+  <li>Let <var>requestedURL</var> be <var>response</var>'s <a for=response>URL list</a>'s first item.
+  <li>If <var>response</var>'s <a for=response>timing allow passed flag</a> is not set, set
+  <var>timingInfo</var> to a new <a for=/>fetch timing info</a>, with its
+  <a for="fetch timing info">fetch start time</a> set to <var>timingInfo</var>'s
+  <a for="fetch timing info">fetch start time</a>,
+  <a for="fetch timing info">response end time</a> set to <var>timingInfo</var>'s
+  <a for="fetch timing info">response end time</a>,
+  <a for="fetch timing info">connection timing info</a> set to a new
+  <a for=/>connection timing info</a> with its
+  <a for="connection timing info">alpn negotiated protocol</a> set to <var>timingInfo</var>'s
+  <a for="fetch timing info">connection timing info</a>'s
+  <a for="connection timing info">alpn negotiated protocol</a>.
```

Comment:
This is always leaked? That's a bit surprising.

Reply:
I agree. Ongoing discussion about this: https://github.com/w3c/resource-timing/pull/224
Trying to align fetch with resource-timing before making changes to the content.